logstash-core/src/test/java/org/logstash/config/ir/EventConditionTest.java

```diff
@@ -0,0 +1,131 @@
+package org.logstash.config.ir;
+
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.MatcherAssert;
+import org.jruby.RubyArray;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.logstash.RubyUtil;
+import org.logstash.ext.JrubyEventExtLibrary;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedTransferQueue;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import static org.logstash.config.ir.CompiledPipelineTest.IDENTITY_FILTER;
+import static org.logstash.ext.JrubyEventExtLibrary.RubyEvent;
+
+public final class EventConditionTest extends RubyEnvTestCase {
+
+    /**
+     * Globally accessible map of test run id to a queue of {@link JrubyEventExtLibrary.RubyEvent}
+     * that can be used by Ruby outputs.
+     */
+    private static final Map<Long, Collection<JrubyEventExtLibrary.RubyEvent>> EVENT_SINKS =
+            new ConcurrentHashMap<>();
+
+    private static final AtomicLong TEST_RUN = new AtomicLong();
+
+    /**
+     * Unique identifier for this test run so that mock test outputs can correctly identify
+     * their event sink in {@link #EVENT_SINKS}.
+     */
+    private long runId;
+
+    @Before
+    public void beforeEach() {
+        runId = TEST_RUN.incrementAndGet();
+        EVENT_SINKS.put(runId, new LinkedTransferQueue<>());
```

Comment:
Why not make the sink `Collection` a property of the test and send it `Collection#clear` in the test's after hook? 

It's not immediately clear how a static registry of per-instance collections adds value here, especially since we are accessing it from within a closure that has instance properties in scope (e.g., it uses `runId`).

Reply:
This test class uses the same test setup as `CompiledPipelineTest`. With a single test, it doesn't matter, but I left it there so it would accommodate additional tests in the future.

Reply:
I still fail to see the value in this layer of indirection. If we are overriding the property `runId` for each run _and relying on it remaining unchanged during the course of executing a test method_, what value does a static `EVENT_SYNCS` that indexes sync `Collection` per `runId`? Why not just have a sync `Collection` property directly?